### DISCUSSION TITLE
Client not appearing after *supposedly* connecting agent

### DISCUSSION THREAD

**Discussion by Bluscrn99** [Ylianst/MeshCentral#6560] [Issues] [answered]
My issue is that after attempting to connect my mesh agent with no apparent error, the client does not appear on my server

**To Reproduce,**
Host a server locally, and connect to it like: 123.456.78.90:port
Make a computer group
Get the windows x64 agent app (for that group)
Press install, then relaunch and press connect
The client does not appear.

**Expected behavior,**
Host a server locally, and connect to it like: 123.456.78.90:port
Make a computer group
Get the windows x64 agent app (for that group)
Press install, then relaunch and press connect
see client on list and connect.

**Screenshots**
![image](https://github.com/user-attachments/assets/b1fd4087-c7ec-418e-81ef-28d80cf5707c)
![image](https://github.com/user-attachments/assets/3a965aaf-9e4c-450a-9898-9b712e2eb6dc)

**Server Software:**
 - OS: Windows (Server & Client)
 - Virtualization: ?
 - Network: Home Network??
 - Version: 1.1.33
 - Node: 22.11.0

**Client Device**
 - Device: Tower
 - OS: Windows
 - Network: Same Network (for now)
 - Browser: Opera

**Remote Device**
They are all the same device, for now.
**Your config.json file**
```
{
  "$schema": "https://raw.githubusercontent.com/Ylianst/MeshCentral/master/meshcentral-config-schema.json",
  "__comment1__": "This is a simple configuration file, all values and sections that start with underscore (_) are ignored. Edit a section and remove the _ in front of the name. Refer to the user's guide for details.",
  "__comment2__": "See node_modules/meshcentral/sample-config-advanced.json for a more advanced example.",
  "settings": {
    "_cert": "myserver.mydomain.com",
    "_WANonly": true,
    "_LANonly": true,
    "_sessionKey": "MyReallySecretPassword1",
    "_port": 443,
    "_aliasPort": 443,
    "_redirPort": 80,
    "_redirAliasPort": 80
  },
  "domains": {
    "": {
      "title": "Blu's Server",
      "title2": "mischievous doo-hickeys and what not",
      "_minify": true,
      "_newAccounts": true,
      "_userNameIsEmail": true
    }
  },
  "_letsencrypt": {
    "__comment__": "Requires NodeJS 8.x or better, Go to https://letsdebug.net/ first before trying Let's Encrypt.",
    "email": "myemail@mydomain.com",
    "names": "myserver.mydomain.com",
    "skipChallengeVerification": true,
    "production": false
  }
}
```

**Comment by si458** [comment]
Ok, so for starters, you only need to use either install or connect when u run the meshagent, u don't need both

Also if ur config.json is correct, u haven't done any configuration, so meshcentral will run in lan mode, which means only machines on the same network/subnet can connect to ur meshcentral
(You can verify this by running meshagent again and clicking connection info and checking its url and it should show LOCAL)

**Comment by Bluscrn99** [comment]
How do I turn off lan mode? I was thinking I would connect by turning on port forwarding and doing ip:port remotely.

**Comment by si458** [comment, marked as answer]
@TheIdiot0 well the correct way to do it in your case is the following
1. create a DNS A record and point it to your external IP address
2. then change `_cert` to `cert` and set the DNS name to it `"cert": "my.exampledomain.com"`
3. then restart meshcentral
4. make sure you can access the web ui using the DNS name
5. then change `_letsencrypt` to `letsencrypt` and fill in the `letsencrypt` section (email, names)
6. restart meshcentral,
7. wait about 2mins, then visit your url again, and you should get the letsencrypt staging certificate and the login screen
8. one final time, change `production` from `false` to `true` inside of `letsencrypt`
9. restart meshcentral
10. wait again about 2mins, then visit your DNS name and you should have a VALID certificate 👍

**Reply by Bluscrn99** [reply]
So, would this fix my issue with the agent connection?

**Reply by Bluscrn99** [reply]
Worked!
